src/openrct2/rct2/SeaDecrypt.cpp

```diff
@@ -0,0 +1,96 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2020 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "../common.h"
+#include "../core/File.h"
+#include "../core/Path.hpp"
+#include "RCT2.h"
+
+#include <cstdint>
+#include <memory>
+#include <string_view>
+
+constexpr int32_t MASK_SIZE = 0x1000;
+
+struct EncryptionKey
+{
+    uint32_t Seed0{};
+    uint32_t Seed1{};
+};
+
+static EncryptionKey GetEncryptionKey(const std::string_view& fileName)
```

Comment:
So if user renames the file, it won't decrypt?

Reply:
Have you read the PR description?

Reply:
Obviously I haven't. Thanks for bringing it to my attention.